nova/virt/vmwareapi/vmops.py

```diff
@@ -3125,3 +3129,97 @@ def update_vmref_cache(self):
                 continue
 
             vm_util.vm_ref_cache_update(vm_uuid, props['obj'])
+
+    def sync_server_group(self, context, sg_uuid):
+        LOG.debug('Starting sync for server-group %s', sg_uuid)
+
+        @utils.synchronized('vmware-server-group-{}'.format(sg_uuid))
+        def _sync_sync_server_group(self, context, sg_uuid):
+            rule_prefix = '{}{}'.format(constants.DRS_PREFIX, sg_uuid)
+
+            # retrieve the server-group with its members
+            try:
+                sg = objects.instance_group.InstanceGroup.get_by_uuid(context,
+                                                                      sg_uuid)
+            except exception.InstanceGroupNotFound:
+                LOG.info('Server-group %s cannot be found in DB', sg_uuid)
+                # check if we have it in the vCenter. if yes, it's an orphan
+                # and needs deletion
+                rules = cluster_util.get_rules_by_prefix(
+                    self._session, self._cluster, rule_prefix)
+                for rule in rules:
+                    LOG.debug('Deleting DRS rule %s as orphan', rule.name)
+                    cluster_util.delete_vm_rule(
+                        self._session, self._cluster, rule)
+                    LOG.info('Deleted rule %s as orphan', rule.name)
+                LOG.debug('Sync for server-group %s done', sg_uuid)
+                return
+
+            # retrieve the instances, because sg.members contains all members
+            # and we need to filter them for our host
+            InstanceList = objects.instance.InstanceList
+            filters = {'host': self._virtapi._compute.host, 'uuid': sg.members,
```

Comment:
`_virtapi._compute.host` goes through a private member. I rather would like to avoid that

Reply:
I'd like to avoid that, too. The only way, I see, is to pass the `_compute_host` on to the `_vmops` in the driver, once we get it set. Didn't sound that great either, but if you prefer that, we can change it.

Reply:
Well, I think technically it would be the cleanest solution. I would suggest to start the sync loop in the `ComputeDriver.init_host` function call and pass the host there to the loop.
It would also ensure, that the loop will only be started after the driver has been initialised.

Reply:
Ok, so we save the host onto the vmops object but not the driver object, because we don't need it on the driver-object anymore.

It looks like `init_host()` is one of the first things, that are getting called on nova-compute startup - and `driver.init_host()` is pretty much the first thing in there. This means, we should be fine with `sync_server_group()` needing `self._compute_host` to be set on the vmops object.

Reply:
Sure, I would be fine with that.